### ISSUE TITLE
wrapper

### ISSUE THREAD

**Issue by Worldwide7477** [Moc/wrapper#4]
Not really understanding the wrapper?
I'm trying to implant my website within e107 when i open the plugin, I name it as my site name i use a full url that points to website folder, i also use HEIGHT and WIDTH using blank - for both no scroll bars and user class everybody.
After trying to create it never finishes getting sql errors maybe i'm doing something wrong below are errors that shows each time:

```
Action: create

Array
(
    [e-token] => ace62ceee26b877eede590a441e2dada
    [wrapper_title] => TriCityTorrents
    [wrapper_url] => http://tricitytorrents.com/rasta/
    [wrapper_height] => 
    [wrapper_width] => 
    [wrapper_scrollbars] => 0
    [wrapper_userclass] => 0
    [etrigger_submit] => create
    [__after_submit_action] => list
    [submit_value] => 0
    [mode] => 
)

Executing _blank_admin :: process()
SQL Error #0:
SQL QRY Error

Array
(
    [_FIELD_TYPES] => Array
        (
            [wrapper_title] => str
            [wrapper_url] => str
            [wrapper_height] => str
            [wrapper_width] => str
            [wrapper_scrollbars] => int
            [wrapper_userclass] => int
        )

    [data] => Array
        (
            [wrapper_id] => 0
            [wrapper_title] => TriCityTorrents
            [wrapper_url] => http://tricitytorrents.com/rasta/
            [wrapper_height] => 
            [wrapper_width] => 
            [wrapper_scrollbars] => 0
            [wrapper_userclass] => 0
        )

)
```

**Comment by Moc** [comment]
Thanks for posting up about your issue. I'll get back to you asap.

**Comment by Worldwide7477** [comment]
Thanks Moc looking forward to it.

**Comment by Moc** [comment]
I'm in the process of testing this. Thus far I have no luck reproducing the issue. 
- Are you using the latest version of e107? You can try to update from Github (Admin Area > Tools > Database > Update from Github) 
- Which PHP and MySQL version are you running? (Admin Area > About > PHP Info > left sidebar) 

```
SQL Error #0:
SQL QRY Error
```

This indicates an issue with the MySQL server connection, but this is weird since the rest of your e107 installation I assume works fine. 

Looking into this further asap .

**Comment by Worldwide7477** [comment]
Hi Tijn,
This is what I have:
_e107 *Version 2.1.2 (git), *PHP Version_5.5.37, *MySQL *
5.5.5-10.0.26-MariaDB

I have noticed on some plugins they will not download like yours i had to
do manually, and i do think that there is a bug in e107 as I had Cameron
look at it as well I really do not know whats going on if I try to download
a plugin it throws a error - " is not in plugins DB and do not know what it
could be as i checked to make sure everything from the sql was on the
DB....i also checked to make sure curl was installed as well as curl
extension=curl.so
was in php.ini.

Asso says no updates

This is driving me mad.

**Comment by Moc** [comment]
Hmm, maybe there is a bug with MariaDB, even though this should work just fine. 
- Do you have MySQL running in strict mode?

**Comment by Worldwide7477** [comment]
i will have look up and how its done through cpanel...never checked that

**Comment by Worldwide7477** [comment]
I will have to look up and see how its done through cpanel...never checked
that

**Comment by Worldwide7477** [comment]
just checked it was on and i turn it off by adding this to my.cnf file:

sql_mode=”NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION”

I restarted the apache went to see if any would download from plugin
area....and still "NO GO"

I might have to redo my complete server cause I really have no clue to
whats going on....I don't know if its on e-107 side or if it has something
to do with curl.

**Comment by Moc** [comment]
This is currently being discussed on Gitter. There is a new fix on the Github of e107 (core), which may help with your issue. 
- On which port is your MySQL server running?

**Comment by Moc** [comment]
I have found some bugs in my code, some of which may be related to your issue. I'll provide updates asap.

**Comment by Worldwide7477** [comment]
Thanks Tijn mysql is running port 3306, hope you can fix it...in the mean
time i will work on a few plugins to integrate that way as well just in
case i still have problems later.

**Comment by Moc** [comment]
Just fyi, I expect to have the update released by the end of this week :)

**Comment by Worldwide7477** [comment]
Thank you Tijn look forward to it.
Still trying to figure out why downloads other then e107 will not download
for plugins as well as theme....but if its a e107 plugin or theme it will
install.
As well as the github update will not download the zip file. kinda weird.

Worldwide7477

**Comment by Moc** [comment]
Some delays, but I have not forgotten you. Will take a look at this again asap (I have a lot of things going on atm).

**Comment by Worldwide7477** [comment]
No Problem Moc, i'm working on plugins for a standalone page as well so if
it does not work then i will use your wrap when its done.I'm hoping to get
a fully integrate of the site going this way but if it gets to crazy then i
will just wrap it.

**Comment by Moc** [comment]
Sorry for the incredible delay. Life came looking around the corner and I had to dedicate my time to other things. I have recently updated the Wrapper plugin and I'll be releasing the 1.1.0 version over the next few days. 

Could you please test it for me again? Instructions: 

- Update e107 from Github (Admin Area > Database > Tools > Sync with Github) 
- Update the wrapper plugin using the latest version from Github. Download the files here: https://github.com/Moc/wrapper/archive/master.zip
- Go the home of the Admin Area and you should be asked to perform a database update for the Wrapper plugin. Please execute this update. 
- Try to use the Wrapper plugin again, and see if the error is still occurring for you. Please provide feedback here.

**Comment by Moc** [comment]
I am considering this fixed in the next release: 1.1.0. 
Please re-open in case this issue is still present in the latest release.
